src/numpy-stubs/_typing/_ufunc.pyi

```diff
@@ -222,6 +222,72 @@ class _Call11Bool(Protocol):
     ) -> NDArray[np.bool] | np.bool: ...
 
 @type_check_only
+class _Call11Logical(Protocol):
+    @overload
+    def __call__(  # (scalar, dtype: np.object_) -> bool
+        self,
+        x: _ScalarLike_co,
+        /,
+        dtype: _DTypeLike[np.object_],
+        out: None = None,
```

Comment:
I just noticed that `dtype` is keyword-only according to the docstring:

```
logical_not = <ufunc 'logical_not'>
    logical_not(x, /, out=None, *, where=True, casting='same_kind', order='K', dtype=None, subok=True[, signature])
```

i.e.

```suggestion
        out: None = None,
        *,
        dtype: _DTypeLike[np.object_],
```

Reply:
also, be careful with reordering parameters that aren't keyword-only